source/Halibut.TestUtils.CompatBinary.Base/Services/StayAliveUntilHelper.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Halibut.TestUtils.SampleProgram.Base.Services
+{
+    public class StayAliveUntilHelper
+    {
+        public static void WaitUntilSignaledToDie()
+        {
+            var stayAliveFile = SettingsHelper.CompatBinaryStayAliveLockFile();
```

Comment:
How does this work if we run multiple tests in parallel? As they would all use the same directory, right? Would that not cause issue?

Reply:
This binary is started by a test in a block like this:
```
            using (var clientAndService = await PreviousClientVersionAndServiceBuilder.ForServiceConnectionType(serviceConnectionType)
                       .WithClientVersion(PreviousVersions.v5_0_429)
                       .WithEchoServiceService(echoService)
                       .Build())
```

Multiple tests in parallel would be starting multiple binaries each with which would get their own unique file.